# Avance 1 

### Equipo 2

* Antonio Mendoza Lopez
* Jesus Mario Sauceda Flores

### Objetivo.

Se planea saber en que rangos de edades se suicidan más las personas en México.

### Preguntas de Interés.
* ¿En que rangos de edades son más comunes los suicidios?
* ¿La edad influye en el numero de suicidios?

### Introducción.

La base de datos esta orientada al numero de suicidios que han ocurrido desde 1985 hasta 2016, y viene categorizado por el país donde ocurren los suicidios, el rango de edad, numero de habitantes, sexo, ingreso per capito, producto interno bruto de país, generacio y suicidios por cada 100,000 habitantes, en lo que nos enfocaremos principalmente son en los rangos de edades, ya que queremos saber si la edad afecta en la cantidad de suicidos que existen.

### Procesamiento de los Datos.

Importar las librerías.

In [43]:
import pandas as pd 
import numpy as np
import matplotlib as plt
import matplotlib.pyplot as plt
%matplotlib inline

Cargamos nuestra base de datos.

In [44]:
suicidios=pd.read_csv("/Users/antoniomendoza/Desktop/Mineria/suicidios.csv")
datos=pd.DataFrame(suicidios)
datos.head()

country  year     sex          age  suicides_no  population  \
0  Albania  1987    male  15-24 years           21      312900   
1  Albania  1987    male  35-54 years           16      308000   
2  Albania  1987  female  15-24 years           14      289700   
3  Albania  1987    male    75+ years            1       21800   
4  Albania  1987    male  25-34 years            9      274300   

   suicides/100k pop country-year  HDI for year  gdp_for_year ($)   \
0               6.71  Albania1987           NaN      2,156,624,900   
1               5.19  Albania1987           NaN      2,156,624,900   
2               4.83  Albania1987           NaN      2,156,624,900   
3               4.59  Albania1987           NaN      2,156,624,900   
4               3.28  Albania1987           NaN      2,156,624,900   

   gdp_per_capita ($)       generation  
0                 796     Generation X  
1                 796           Silent  
2                 796     Generation X  
3                 796  G.I. Generation  
4                 796          Boomers

In [55]:
datos.shape

(27820, 12)

In [54]:
datos.dtypes

country                object
year                    int64
sex                    object
age                    object
suicides_no             int64
population              int64
suicides/100k pop     float64
country-year           object
HDI for year          float64
 gdp_for_year ($)      object
gdp_per_capita ($)      int64
generation             object
dtype: object

Tenemos un total de 27,820 filas pero nosotros solo requerimos las que incluyen a México, de igual manera no son necesarias todas las columnas solo dejaremos, el año el sexo, rango de edad, población y generación.

In [46]:
daton=datos.drop(["gdp_per_capita ($)","suicides/100k pop","country-year","HDI for year"," gdp_for_year ($) "],axis=1)


Ahora tenemos la base de datos modificada, ahora solo falta mostrar los datos de México.

In [48]:
daton.head()

country  year     sex          age  suicides_no  population  \
0  Albania  1987    male  15-24 years           21      312900   
1  Albania  1987    male  35-54 years           16      308000   
2  Albania  1987  female  15-24 years           14      289700   
3  Albania  1987    male    75+ years            1       21800   
4  Albania  1987    male  25-34 years            9      274300   

        generation  
0     Generation X  
1           Silent  
2     Generation X  
3  G.I. Generation  
4          Boomers

In [50]:
solomex=daton[daton["country"]=="Mexico"]

In [51]:
solomex.head()

country  year   sex          age  suicides_no  population  \
16564  Mexico  1985  male    75+ years           44      432000   
16565  Mexico  1985  male  55-74 years          145     2330000   
16566  Mexico  1985  male  25-34 years          340     5679000   
16567  Mexico  1985  male  35-54 years          327     5836000   
16568  Mexico  1985  male  15-24 years          375     8420000   

            generation  
16564  G.I. Generation  
16565  G.I. Generation  
16566          Boomers  
16567           Silent  
16568     Generation X

In [53]:
solomex.shape

(372, 7)

Ahora tenemos la base de datos modificada con los datos que necesitamos para nuestro proyecto.